src/main/c/internal/ZlibContext.h

```diff
@@ -0,0 +1,103 @@
+#pragma once
+
+#include <zlib.h>
+
+#include <vector>
+#include <stdexcept>
+
+
+class ZlibContext {
+public:
```

Comment:
Probably best to make this non-copyable. I'll happily do that in a follow-up change unless you agree and get to it first :)